openedx/core/djangoapps/discussions/serializers.py

```diff
@@ -0,0 +1,265 @@
+        """
+        save = False
+        cohort_settings = {}
+        for field, value in validated_data.items():
+            if field in self.Meta.fields:
+                setattr(instance, field, value)
+                save = True
+            else:
+                cohort_settings[field] = value
+        if cohort_settings:
+            discussion_settings = CourseDiscussionSettings.get(instance.id)
+            serializer = DiscussionSettingsSerializer(
+                discussion_settings,
+                context={
+                    'course': instance,
+                    'settings': discussion_settings,
+                },
+                data=cohort_settings,
+                partial=True,
+            )
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+        if save:
+            modulestore().update_item(instance, self.context['user_id'])
+        return instance
+
+
+class DiscussionsConfigurationSerializer(serializers.ModelSerializer):
+    """
+    Serialize configuration responses
+    """
+
+    class Meta:
+        model = DiscussionsConfiguration
+        fields = [
+            'enabled',
+            'provider_type',
+        ]
+
+    def create(self, validated_data):
+        """
+        We do not need this.
+        """
+        raise NotImplementedError
+
+    def to_internal_value(self, data: dict) -> dict:
+        """
+        Transform the *incoming* primitive data into a native value.
+        """
+        payload = {
+            'context_key': data.get('course_key', ''),
+            'enabled': data.get('enabled', False),
+            'lti_configuration': data.get('lti_configuration', {}),
+            'plugin_configuration': data.get('plugin_configuration', {}),
+            'provider_type': data.get('provider_type', DEFAULT_PROVIDER_TYPE),
+        }
+        return payload
+
+    def to_representation(self, instance: DiscussionsConfiguration) -> dict:
+        """
+        Serialize data into a dictionary, to be used as a response
+        """
+        payload = super().to_representation(instance)
+        lti_configuration_data = {}
+        supports_lti = instance.supports('lti')
+        if supports_lti:
+            lti_configuration = LtiSerializer(instance.lti_configuration)
+            lti_configuration_data = lti_configuration.data
+        provider_type = instance.provider_type or DEFAULT_PROVIDER_TYPE
+        plugin_configuration = instance.plugin_configuration
+        if provider_type == 'legacy':
+            course_key = instance.context_key
+            course = get_course_by_id(course_key)
+            legacy_settings = LegacySettingsSerializer(
+                course,
+                data=plugin_configuration,
+            )
+            if legacy_settings.is_valid(raise_exception=True):
+                plugin_configuration = legacy_settings.data
+        payload.update({
+            'features': list(FEATURES.values()),
```

Comment:
`FEATURES.values()` will return a list, why you need to wrap it inside a `list()` callable?

Reply:
`.values()` on a dict doesn't return a list in python3, only in 2.